inst/stan/marginal_model/functions.stan

```diff
@@ -0,0 +1,10 @@
+real primarycensored_wrapper_lpmf(data int d, real mu, real sigma, data real pwindow) {
+  int dist_id = 1; // lognormal
+  array[2] real params;
+  params[1] = mu;
+  params[2] = sigma;
+  int d_upper = d + 1;
```

Comment:
I'd suggest you pass in d_upper (and swindow for posterior prediction)

Reply:
I was a bit confused about what exactly `d_upper` and `D` are